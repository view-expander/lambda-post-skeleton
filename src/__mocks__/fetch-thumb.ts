import MockAdapter from 'axios-mock-adapter'
import api from '../api'
export const DUMMY_IMAGE = new Uint8Array(
  Buffer.from(
    '/9j/4AAQSkZJRgABAQEASABIAAD/4gxYSUNDX1BST0ZJTEUAAQEAAAxITGlubwIQAABtbnRyUkdCIFhZWiAHzgACAAkABgAxAABhY3NwTVNGVAAAAABJRUMgc1JHQgAAAAAAAAAAAAAAAAAA9tYAAQAAAADTLUhQICAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABFjcHJ0AAABUAAAADNkZXNjAAABhAAAAGx3dHB0AAAB8AAAABRia3B0AAACBAAAABRyWFlaAAACGAAAABRnWFlaAAACLAAAABRiWFlaAAACQAAAABRkbW5kAAACVAAAAHBkbWRkAAACxAAAAIh2dWVkAAADTAAAAIZ2aWV3AAAD1AAAACRsdW1pAAAD+AAAABRtZWFzAAAEDAAAACR0ZWNoAAAEMAAAAAxyVFJDAAAEPAAACAxnVFJDAAAEPAAACAxiVFJDAAAEPAAACAx0ZXh0AAAAAENvcHlyaWdodCAoYykgMTk5OCBIZXdsZXR0LVBhY2thcmQgQ29tcGFueQAAZGVzYwAAAAAAAAASc1JHQiBJRUM2MTk2Ni0yLjEAAAAAAAAAAAAAABJzUkdCIElFQzYxOTY2LTIuMQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWFlaIAAAAAAAAPNRAAEAAAABFsxYWVogAAAAAAAAAAAAAAAAAAAAAFhZWiAAAAAAAABvogAAOPUAAAOQWFlaIAAAAAAAAGKZAAC3hQAAGNpYWVogAAAAAAAAJKAAAA+EAAC2z2Rlc2MAAAAAAAAAFklFQyBodHRwOi8vd3d3LmllYy5jaAAAAAAAAAAAAAAAFklFQyBodHRwOi8vd3d3LmllYy5jaAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABkZXNjAAAAAAAAAC5JRUMgNjE5NjYtMi4xIERlZmF1bHQgUkdCIGNvbG91ciBzcGFjZSAtIHNSR0IAAAAAAAAAAAAAAC5JRUMgNjE5NjYtMi4xIERlZmF1bHQgUkdCIGNvbG91ciBzcGFjZSAtIHNSR0IAAAAAAAAAAAAAAAAAAAAAAAAAAAAAZGVzYwAAAAAAAAAsUmVmZXJlbmNlIFZpZXdpbmcgQ29uZGl0aW9uIGluIElFQzYxOTY2LTIuMQAAAAAAAAAAAAAALFJlZmVyZW5jZSBWaWV3aW5nIENvbmRpdGlvbiBpbiBJRUM2MTk2Ni0yLjEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHZpZXcAAAAAABOk/gAUXy4AEM8UAAPtzAAEEwsAA1yeAAAAAVhZWiAAAAAAAEwJVgBQAAAAVx/nbWVhcwAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAo8AAAACc2lnIAAAAABDUlQgY3VydgAAAAAAAAQAAAAABQAKAA8AFAAZAB4AIwAoAC0AMgA3ADsAQABFAEoATwBUAFkAXgBjAGgAbQByAHcAfACBAIYAiwCQAJUAmgCfAKQAqQCuALIAtwC8AMEAxgDLANAA1QDbAOAA5QDrAPAA9gD7AQEBBwENARMBGQEfASUBKwEyATgBPgFFAUwBUgFZAWABZwFuAXUBfAGDAYsBkgGaAaEBqQGxAbkBwQHJAdEB2QHhAekB8gH6AgMCDAIUAh0CJgIvAjgCQQJLAlQCXQJnAnECegKEAo4CmAKiAqwCtgLBAssC1QLgAusC9QMAAwsDFgMhAy0DOANDA08DWgNmA3IDfgOKA5YDogOuA7oDxwPTA+AD7AP5BAYEEwQgBC0EOwRIBFUEYwRxBH4EjASaBKgEtgTEBNME4QTwBP4FDQUcBSsFOgVJBVgFZwV3BYYFlgWmBbUFxQXVBeUF9gYGBhYGJwY3BkgGWQZqBnsGjAadBq8GwAbRBuMG9QcHBxkHKwc9B08HYQd0B4YHmQesB78H0gflB/gICwgfCDIIRghaCG4IggiWCKoIvgjSCOcI+wkQCSUJOglPCWQJeQmPCaQJugnPCeUJ+woRCicKPQpUCmoKgQqYCq4KxQrcCvMLCwsiCzkLUQtpC4ALmAuwC8gL4Qv5DBIMKgxDDFwMdQyODKcMwAzZDPMNDQ0mDUANWg10DY4NqQ3DDd4N+A4TDi4OSQ5kDn8Omw62DtIO7g8JDyUPQQ9eD3oPlg+zD88P7BAJECYQQxBhEH4QmxC5ENcQ9RETETERTxFtEYwRqhHJEegSBxImEkUSZBKEEqMSwxLjEwMTIxNDE2MTgxOkE8UT5RQGFCcUSRRqFIsUrRTOFPAVEhU0FVYVeBWbFb0V4BYDFiYWSRZsFo8WshbWFvoXHRdBF2UXiReuF9IX9xgbGEAYZRiKGK8Y1Rj6GSAZRRlrGZEZtxndGgQaKhpRGncanhrFGuwbFBs7G2MbihuyG9ocAhwqHFIcexyjHMwc9R0eHUcdcB2ZHcMd7B4WHkAeah6UHr4e6R8THz4faR+UH78f6iAVIEEgbCCYIMQg8CEcIUghdSGhIc4h+yInIlUigiKvIt0jCiM4I2YjlCPCI/AkHyRNJHwkqyTaJQklOCVoJZclxyX3JicmVyaHJrcm6CcYJ0kneierJ9woDSg/KHEooijUKQYpOClrKZ0p0CoCKjUqaCqbKs8rAis2K2krnSvRLAUsOSxuLKIs1y0MLUEtdi2rLeEuFi5MLoIuty7uLyQvWi+RL8cv/jA1MGwwpDDbMRIxSjGCMbox8jIqMmMymzLUMw0zRjN/M7gz8TQrNGU0njTYNRM1TTWHNcI1/TY3NnI2rjbpNyQ3YDecN9c4FDhQOIw4yDkFOUI5fzm8Ofk6Njp0OrI67zstO2s7qjvoPCc8ZTykPOM9Ij1hPaE94D4gPmA+oD7gPyE/YT+iP+JAI0BkQKZA50EpQWpBrEHuQjBCckK1QvdDOkN9Q8BEA0RHRIpEzkUSRVVFmkXeRiJGZ0arRvBHNUd7R8BIBUhLSJFI10kdSWNJqUnwSjdKfUrESwxLU0uaS+JMKkxyTLpNAk1KTZNN3E4lTm5Ot08AT0lPk0/dUCdQcVC7UQZRUFGbUeZSMVJ8UsdTE1NfU6pT9lRCVI9U21UoVXVVwlYPVlxWqVb3V0RXklfgWC9YfVjLWRpZaVm4WgdaVlqmWvVbRVuVW+VcNVyGXNZdJ114XcleGl5sXr1fD19hX7NgBWBXYKpg/GFPYaJh9WJJYpxi8GNDY5dj62RAZJRk6WU9ZZJl52Y9ZpJm6Gc9Z5Nn6Wg/aJZo7GlDaZpp8WpIap9q92tPa6dr/2xXbK9tCG1gbbluEm5rbsRvHm94b9FwK3CGcOBxOnGVcfByS3KmcwFzXXO4dBR0cHTMdSh1hXXhdj52m3b4d1Z3s3gReG54zHkqeYl553pGeqV7BHtje8J8IXyBfOF9QX2hfgF+Yn7CfyN/hH/lgEeAqIEKgWuBzYIwgpKC9INXg7qEHYSAhOOFR4Wrhg6GcobXhzuHn4gEiGmIzokziZmJ/opkisqLMIuWi/yMY4zKjTGNmI3/jmaOzo82j56QBpBukNaRP5GokhGSepLjk02TtpQglIqU9JVflcmWNJaflwqXdZfgmEyYuJkkmZCZ/JpomtWbQpuvnByciZz3nWSd0p5Anq6fHZ+Ln/qgaaDYoUehtqImopajBqN2o+akVqTHpTilqaYapoum/adup+CoUqjEqTepqaocqo+rAqt1q+msXKzQrUStuK4trqGvFq+LsACwdbDqsWCx1rJLssKzOLOutCW0nLUTtYq2AbZ5tvC3aLfguFm40blKucK6O7q1uy67p7whvJu9Fb2Pvgq+hL7/v3q/9cBwwOzBZ8Hjwl/C28NYw9TEUcTOxUvFyMZGxsPHQce/yD3IvMk6ybnKOMq3yzbLtsw1zLXNNc21zjbOts83z7jQOdC60TzRvtI/0sHTRNPG1EnUy9VO1dHWVdbY11zX4Nhk2OjZbNnx2nba+9uA3AXcit0Q3ZbeHN6i3ynfr+A24L3hROHM4lPi2+Nj4+vkc+T85YTmDeaW5x/nqegy6LzpRunQ6lvq5etw6/vshu0R7ZzuKO6070DvzPBY8OXxcvH/8ozzGfOn9DT0wvVQ9d72bfb794r4Gfio+Tj5x/pX+uf7d/wH/Jj9Kf26/kv+3P9t////2wCEAAIDAwMEAwQFBQQGBgYGBggIBwcICA0JCgkKCQ0TDA4MDA4MExEUEQ8RFBEeGBUVGB4jHRwdIyolJSo1MjVFRVwBAgMDAwQDBAUFBAYGBgYGCAgHBwgIDQkKCQoJDRMMDgwMDgwTERQRDxEUER4YFRUYHiMdHB0jKiUlKjUyNUVFXP/CABEIAQABAAMBIgACEQEDEQH/xAAeAAEBAQEAAgMBAQAAAAAAAAAACQcKBAYDBQgBAv/aAAgBAQAAAADn/AAbLQSlNBdlAAcAYADZaCUpoLsoADgDAAbLQSlNBdlAAcAYADZaCUpoLsoADgDAAbLQSlNBdlAAcAYADZaCUpoLsoADgDAAbLQSlNBdlAAcAYADZaCUpoLsoADgDAAbLQSlNBdlAAcAYADZaCUpoLsoADgDAAbDQellCNgAAcAYADSqKVAoxowADgDAAe60dqxSP28ABwBgANCovUmjvvoADhoAHx/T/Vedun7R/d37R3nzvp/pviAHOkAP8+L4fke26Zq2me2/P4ni/wCQB/QB/fL8v4PUczynM/Ufg8vy/wCgDouAHyfc/c+Bg34v/CP4uwvwvtft/kAHcuAA9BnJLac+egAO/wAAAeoTclPOH0sAB3+AAM4nTL6dWbAAO/wABj0+Jpz4x4AB3+AAMan3NWfeNAAO/wAAAY1Puas+8aAAd/gADGp9zVn3jQADv8AAY1Puas+8aAAd/gADGp9zVn3jQADv8AAY1Puas+8aAAd/gADGp9zVn3jQADv8AAY1Puas+8aAAd/gADGp9zVn3jQADv8AAAGNT7mrPvGgAH//xAAUAQEAAAAAAAAAAAAAAAAAAAAA/9oACAECEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/8QAFAEBAAAAAAAAAAAAAAAAAAAAAP/aAAgBAxAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP/EADIQAAAEAwcCBQQCAwEAAAAAAAABAgYDFqEEMTJSU3GBEUAFBxIhIhMzQ1EwYSNCkWL/2gAIAQEAAT8A/mauNG4Y98PgMi6HwGrgRt3rVxo3DHvh8BkXQ+A1cCNu9auNG4Y98PgMi6HwGrgRt3rVxo3DHvh8BkXQ+A1cCNu9auNG4Y98PgMi6HwGrgRt3rVxo3DHvh8BkXQ+A1cCNu9auNG4Y98PgMi6HwGrgRt3rVxo3DHvh8BkXQ+A1cCNu9auNG4Y98PgMi6HwGrgRt3rVxo3DHvh8BkXQ+A1cCNu9aqy9SAxz94fAY5e0PgNVB+lHHetaN0jEX9hhr6nD4DDR1KFwGnB/wAaT/Rd621GVrLch5fn1OFwPL8upQuA10kVl6/+e9bFmWq0ErpeYYUI0nC4DBikkoXAatqQcEk9by/nlVeQxKq8hiVV5DEqryGJVXkMSqvIYlVeQxKq8hiVV5DEqryGFtVfTAIzWjFcRkFNy2Edx/8ABZmxaFLL1EZhvN5UNSPgGnYzhGj2/QaVsKEUP3/QbzhTDSj5iyuqCaC9RkYU6LKRf6iK64RF0JREFupJnjqJqRnITUjOQmpGchNSM5CakZyE1IzkJqRnITUjOQmpGchNSM5CSD06CSD06CSD06CSD06CSD06CSD06CSD06CSD06CSD06CSD06A2OenQLYZn+KgPy/M/xUEJgmk/tUFjaZwjL4UHh/h/0OnsPD/EPodPcWN2lCIvnUQn8SS+7UH5gFq1C34R/lqDfBalRO5alRO5alRO5alRO5alRO5alRO5alRO5alRO5alRO5alRO5alRI5adBI5adBI5adBI5adBI5adBI5adBI5adBI5adBI5adBI5adATHLToEMMj/HQF5fkZfaoIrCJJfaoLY0yhEfwoPEPD/odfYeIeIfQ6+4tjsOEZ/Ooiv00n92oLzAM/wAtQh+Gf5KgnwepUTwepUTwepUTwepUTwepUTwepUTwepUTwepUTwepUTwepUTwepUSqjIQlVGQhKqMhCVUZCEqoyEJVRkISqjIQlVGQhKqMhCVUZCCGonr9uggtSGRdTSRBLXspF/qLU1YJoP0kRhwt1MNK/gHZYyhEv2/YdtsOEa/f9hwuFUNS/mLS57QpZ+kzMJclrI7z/6ITpjFeowh1K6YxNS85ial5zE1LzmJqXnMTUvOYmpecxNS85ial5zE1LzmJqXnP+d1WVBwTV0vIP2ESSi8h/RTScXkOe0rVaDT1vPvXQoisvT+h5gGRlF5HmARmcXkONJlbD3PvXZGL6ai/oPxfUonIfiOpxeQ6YPSMZ9Lj711LP0r5D4P2ich8F7xOQ6kF618966sC9g+LonIfF8TkOrGvfvXVgXsHxdE5D4vich1Y17966sC9g+LonIfF8TkOrGvfvXVgXsHxdE5D4vich1Y17966sC9g+LonIfF8TkOrGvfvXVgXsHxdE5D4vich1Y17966sC9g+LonIfF8TkOrGvfvXVgXsHxdE5D4vich1Y17966sC9g+LonIfF8TkOrGvfvXVgXsHxdE5D4vich1Y17/AM//xAAUEQEAAAAAAAAAAAAAAAAAAACA/9oACAECAQE/AAB//8QAFBEBAAAAAAAAAAAAAAAAAAAAgP/aAAgBAwEBPwAAf//Z',
    'base64'
  ).buffer
).buffer

export default jest.fn().mockImplementation(async (key: string) => {
  const mock = new MockAdapter(api)
  mock
    .onGet(`source/${key}`)
    .reply(200, DUMMY_IMAGE, { 'content-type': 'image/jpeg' })

  const res = await api.get(`/source/${key}`, {
    params: {
      size: 'thumb',
    },
    responseType: 'arraybuffer',
  })

  mock.restore()

  return Promise.resolve(res)
})
