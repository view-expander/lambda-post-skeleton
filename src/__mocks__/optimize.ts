export const DUMMY_OPTIMIZED_IMAGE = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256"><path fill="#ccc" fill-rule="evenodd" d="M0 66.096v66.096l57.25-.305c63.71-.339 61.754-.53 65.442 6.38 1.587 2.973 1.701 7.828 1.421 60.483l-.305 57.25h131.947l.729-125.25c.401-68.888.524-126.6.273-128.25-.252-1.65-.525 25.013-.607 59.25L256 124l-46.25.25c-65.578.354-67.778.304-71.531-1.637-6.828-3.531-6.652-1.717-6.33-65.363l.29-57.25H0v66.096m.472.404c0 36.3.125 51.005.278 32.677.152-18.327.152-48.027-.001-66C.597 15.205.472 30.2.472 66.5"/></svg>`

export default jest.fn().mockResolvedValue(DUMMY_OPTIMIZED_IMAGE)
