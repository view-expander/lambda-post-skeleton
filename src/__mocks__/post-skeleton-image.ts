export const DUMMY_POST_RESPONSE = {
  $response: {
    httpResponse: {
      statusCode: 200,
    },
  },
}
export default jest.fn().mockResolvedValue(DUMMY_POST_RESPONSE)
