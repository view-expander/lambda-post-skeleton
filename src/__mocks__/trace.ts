export default jest
  .fn()
  .mockResolvedValue(
    '<svg width="8" height="8" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#clip0)"><path fill="#fff" d="M0 0h8v8H0z"/><path fill="#000" d="M0 0h4v4H0zM4 4h4v4H4z"/></g><defs><clipPath id="clip0"><path fill="#fff" d="M0 0h8v8H0z"/></clipPath></defs></svg>'
  )
