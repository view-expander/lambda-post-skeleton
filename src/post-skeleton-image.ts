import * as AWS from 'aws-sdk'
const s3 = new AWS.S3()

export default async (svg: string, key: string) => {
  try {
    const Bucket = process.env.BUCKET

    if (Bucket === undefined) {
      throw new Error('Bucket name is required')
    }

    return s3
      .putObject({
        Bucket,
        Key: `skeleton/${key}`,
        Body: svg,
        ContentType: 'image/svg+xml',
      })
      .promise()
  } catch (err) {
    throw err
  }
}
