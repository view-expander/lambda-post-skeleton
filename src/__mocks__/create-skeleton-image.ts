import { DUMMY_OPTIMIZED_IMAGE } from './optimize'
export default jest.fn().mockResolvedValue(DUMMY_OPTIMIZED_IMAGE)
