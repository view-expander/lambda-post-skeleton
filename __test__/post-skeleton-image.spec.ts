import * as AWS from 'aws-sdk'
import postSkeletonImage from '../src/post-skeleton-image'
import { DUMMY_POST_RESPONSE } from '../src/__mocks__/post-skeleton-image'
import { DUMMY_OPTIMIZED_IMAGE } from './../src/__mocks__/optimize'

describe('postSkeletonImage()', () => {
  beforeEach(() => {
    jest
      .spyOn((AWS.S3 as any).services['2006-03-01'].prototype, 'putObject')
      .mockImplementation(() => ({
        promise: (): any => DUMMY_POST_RESPONSE,
      }))
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('will be succeed', async () => {
    expect.assertions(1)
    return expect(
      postSkeletonImage(DUMMY_OPTIMIZED_IMAGE, 'image.svg')
    ).resolves.toMatchSnapshot()
  })
})
