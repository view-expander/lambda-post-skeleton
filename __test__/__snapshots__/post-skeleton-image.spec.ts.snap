// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postSkeletonImage() will be succeed 1`] = `
Object {
  "$response": Object {
    "httpResponse": Object {
      "statusCode": 200,
    },
  },
}
`;
